src/general_functions.jl

```diff
@@ -328,9 +328,18 @@ function n_th(ω::Real, T::Real)::Float64
     return 1 / (exp(ω / T) - 1)
 end
 
+@doc raw"""
+    tracedist(A::QuantumObject, B::QuantumObject)
 
+Calculates the [trace distance](https://en.wikipedia.org/wiki/Trace_distance) between two [`QuantumObject`](@ref) `A` and `B`:
+``T(A, B) = frac{1}{2} \lVert A - B \rVert_1``
 
-
+Note that `A` and `B` must be either [`Ket`](@ref) or [`Operator`](@ref).
+"""
+tracedist(A::QuantumObject{<:AbstractArray{T1},OperatorQuantumObject}, B::QuantumObject{<:AbstractArray{T2},OperatorQuantumObject}) where {T1,T2} = 0.5 * norm(A - B, 1)
```

Comment:
Should we avoid the 0.5 and directly dividing by 2? In this way we avoid to promote it to Float64, if not. Am I wrong?

Reply:
But usually the output of `tracedist` is `Float`

Reply:
I see, maybe we just change it here. (updated in https://github.com/albertomercurio/QuantumToolbox.jl/pull/102/commits/ef01d3b489b625dc470802dde1125c075ec6c788)